server/app/views/api/ApiDocsView.java

```diff
@@ -0,0 +1,270 @@
+
+    fullProgramDiv.with(leftSide);
+    fullProgramDiv.with(rightSide);
+
+    divTag.with(fullProgramDiv);
+
+    return divTag;
+  }
+
+  private DivTag apiResponseSampleDiv(ProgramDefinition programDefinition) {
+    DivTag apiResponseSampleDiv = div();
+
+    List<QuestionDefinition> questionDefinitions =
+        programDefinition.streamQuestionDefinitions().collect(toImmutableList());
+
+    CfJsonDocumentContext sampleJson = new CfJsonDocumentContext();
+    for (QuestionDefinition questionDefinition : questionDefinitions) {
+      sampleJson.mergeFrom(
+          questionJsonSamplerFactory
+              .create(questionDefinition.getQuestionType())
+              .getSampleJson(questionDefinition));
+    }
+
+    apiResponseSampleDiv.with(
+        pre(code(sampleJson.asPrettyJsonString()))
+            .withStyle("background-color: lightgray;")
+            .withClass("m-4"));
+
+    return apiResponseSampleDiv;
+  }
+
+  private DivTag programDocsDiv(ProgramDefinition programDefinition) {
+    DivTag programDocsDiv = div().withClasses("flex", "flex-col", "gap-4");
+
+    for (QuestionDefinition questionDefinition :
+        programDefinition.streamQuestionDefinitions().collect(toImmutableList())) {
+      programDocsDiv.with(questionDocsDiv(questionDefinition));
+    }
+
+    return programDocsDiv;
+  }
+
+  private DivTag questionDocsDiv(QuestionDefinition questionDefinition) {
+    DivTag divTag = div().withClasses("pl-4", "border", "border-gray-500", "rounded-lg");
+
+    divTag.with(
+        span(
+            h3(b(questionDefinition.getName())).withClasses("inline"),
+            code(" (" + questionDefinition.getQuestionNameKey().toLowerCase(Locale.US) + ")")));
+    divTag.with(br(), br());
+
+    divTag.with(h3("Question Type: " + questionDefinition.getQuestionType().toString()));
+    try {
+      divTag.with(
+          span(
+              h3(text("Question Text:  ")).withClasses("inline"),
+              blockquote(questionDefinition.getQuestionText().get(Locale.US))
+                  .withClasses("inline")));
+
+    } catch (TranslationNotFoundException e) {
+      logger.error("No translation found for locale US in question text: " + e.getMessage());
+    }
+
+    if (questionDefinition.getQuestionType().isMultiOptionType()) {
+      MultiOptionQuestionDefinition multiOptionQuestionDefinition =
+          (MultiOptionQuestionDefinition) questionDefinition;
+      divTag.with(h3("Question Options: " + getOptionsString(multiOptionQuestionDefinition)));
+    }
+
+    return divTag;
+  }
+
+  private static String getOptionsString(MultiOptionQuestionDefinition questionDefinition) {
+    ImmutableList<LocalizedQuestionOption> options =
+        questionDefinition.getOptionsForDefaultLocale();
+    ImmutableList<String> optionsText =
+        options.stream().map(LocalizedQuestionOption::optionText).collect(toImmutableList());
+
+    StringBuilder stringBuilder = new StringBuilder();
+
+    String commaSpace = ", ";
```

Comment:
does something like this work without having to trim the trailing comma off?
`stringBuilder.append("\"").append(String.join("\", \"", optionsText)).append("\"")`

Reply:
Yep! Done.